lib/ui/painting.dart

```diff
@@ -1649,16 +1649,21 @@ class Codec extends NativeFieldWrapperClass2 {
 /// failed.
 Future<Codec> instantiateImageCodec(Uint8List list, {
   double decodedCacheRatioCap = double.infinity,
+  int maxWidth = -1,
```

Comment:
is maxWidth/Height the right parameter here? This API should probably take a Size instead. 

lets think about the behaviors:

- I want to render a small image into a larger area
- I want to render a large image into a smaller area
- I want to resize a large image into a smaller area
- I want to resize a small image into a larger area (?)

I'm also not sure if this should be tied to the existing imageCodec, would it make more sense to expose a separate method for resizing?

Also, this API doesn't allow us to decode an image without loading it entirely into memory - which was a point brought up on the original problem